internal/access/access.go

```diff
@@ -42,6 +42,8 @@ func RequireInfraRole(c *gin.Context, oneOfRoles ...string) (data.GormTxn, error
 		return nil, fmt.Errorf("no active identity")
 	}
 
+	// TODO: use privilege in (...) instead of iterating over roles
+	// TODO: check for test coverage that uses permission from a group
```

Comment:
The first TODO I think is for a separate PR. The second one might be good to do in this one. I think we do have some test coverage from API tests, but I did not find anything for this function itself.

Reply:
Found the test coverage, and renamed the test to `TestRequireInfraRole_GrantsFromGroupMembership` in the latest commit.

Reply:
The first one is addressed in #3176